### PR TITLE
Added feature switch to default Standard Material's new anisotropy texture to off

### PR DESCRIPTION
**Pull request by gagnus**
# Objective

- Standard Material is starting to run out of samplers (currently uses 13 with no additional features off, I think in 0.13 it was 12).
- This change adds a new feature switch, modelled on the other ones which add features to Standard Material, to turn off the new anisotropy feature by default.

## Solution

- feature + texture define

## Testing

- Anisotropy example still works fine
- Other samples work fine
- Standard Material now takes 12 samplers by default on my Mac instead of 13

## Migration Guide

- Add feature pbr_anisotropy_texture if you are using that texture in any standard materials.

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
This PR is marked as draft: what remains to be done to get this ready to review?

**Comment by gagnus** [comment]
By the way I don't know if I'd describe this as a bug fix @alice-i-cecile if it matters?

**Comment by alice-i-cecile** [comment]
Bumping to 0.15 sorry; there's debate on the best way to do this, a workaround exists and this is the last item in the milestone.

**Comment by alice-i-cecile** [comment]
Added back to the milestone. This may not get in, but we need to make a conscious call on it.

File: crates/bevy_pbr/src/render/pbr_fragment.wgsl [hunk 1]
```diff
@@ -420,6 +420,7 @@ fn pbr_input_from_standard_material(
         //
         // This code comes from the `KHR_materials_anisotropy` spec:
         // <https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Khronos/KHR_materials_anisotropy/README.md#individual-lights>
+#ifdef PBR_ANISOTROPY_TEXTURE_SUPPORTED
```
Comment (IceSentry) [inline crates/bevy_pbr/src/render/pbr_fragment.wgsl:423]: I think this should also be around the texture binding not just it's use.

**Review by IceSentry — APPROVED** [review]
I think the ifdef should also be in the bindings file but other than that LGTM.

Eventually it would be nice to have a runtime solution but for 0.14 I think this is good.

File: crates/bevy_pbr/src/render/pbr_fragment.wgsl (hunk 1 above)
Comment (gagnus) [reply crates/bevy_pbr/src/render/pbr_fragment.wgsl:423]: done

**Comment by gagnus** [comment]
added extra ifdef IceSentry identified

**Comment by mockersf** [comment]
This PR is targeting the release branch and GitHub is not happy when changing the base branch from the UI, there are a lot of conflicts.

@gagnus could you target `main` with your PR?

**Comment by gagnus** [comment]
I really don't know what happened here but I've broken the pull request trying to target main instead of release-0.14.

File: crates/bevy_pbr/src/pbr_material.rs [hunk 2]
```diff
@@ -1074,7 +1075,10 @@ impl AsBindGroupShaderType<StandardMaterialUniform> for StandardMaterial {
             clearcoat: self.clearcoat,
             clearcoat_perceptual_roughness: self.clearcoat_perceptual_roughness,
             anisotropy_strength: self.anisotropy_strength,
-            anisotropy_rotation,
+            anisotropy_rotation: vec2(
+                self.anisotropy_rotation.cos(),
+                self.anisotropy_rotation.sin(),
+            ),
```
Comment (pcwalton) [inline crates/bevy_pbr/src/pbr_material.rs:1081]: Why move this here?

Comment (gagnus) [reply crates/bevy_pbr/src/pbr_material.rs:1081]: Sorry, had originally started wrapping cfg around all the values and this fixed a dead code warning, then noticed that wasn’t the case with clearcoat so I did a bit but accidentally left this. Should make no difference?

**Comment by gagnus** [comment]
@pcwalton hopefully fixed your review comment if you could have a check please.

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**